eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -29,35 +29,92 @@ jobs:
           OSVmImage: 'macOS-10.15'
           vcpkg.deps: 'curl[ssl]'
           VCPKG_DEFAULT_TRIPLET: 'x64-osx'
-          build.args: ' -DTRANSPORT_CURL=ON  -DAZ_PLATFORM_IMPL=POSIX'
-
+          build.args: ' -DTRANSPORT_CURL=ON -DAZ_PLATFORM_IMPL=POSIX'
+        Win_x64_with_iot_samples:
+          OSVmImage: 'windows-2019'
+          vcpkg.deps: 'curl[winssl] cmocka paho-mqtt'
+          VCPKG_DEFAULT_TRIPLET: 'x64-windows-static'
+          CMAKE_GENERATOR: 'Visual Studio 16 2019'
+          CMAKE_GENERATOR_PLATFORM: x64
+          build.args: ' -DTRANSPORT_PAHO=ON'
+          test_type: 'iot'
+        Linux_x64_with_iot_samples:
+          OSVmImage: 'ubuntu-18.04'
+          vcpkg.deps: 'curl[ssl] cmocka paho-mqtt'
+          VCPKG_DEFAULT_TRIPLET: 'x64-linux'
+          build.args: ' -DTRANSPORT_PAHO=ON'
+          test_type: 'iot'
     pool:
       vmImage: $(OSVmImage)
 
     steps:
       - template: /eng/pipelines/templates/steps/vcpkg.yml
         parameters:
           DependenciesVariableName: vcpkg.deps
+          
+      - template: /eng/common/TestResources/deploy-test-resources.yml
+        parameters:
+          ServiceDirectory: '.'
+
+      - pwsh: |
+          # Generate certificate 
+          cd $SOURCES_DIR\sdk\samples\iot\
+          openssl ecparam -out device_ec_key.pem -name prime256v1 -genkey
+          openssl req -new -days 365 -nodes -x509 -key device_ec_key.pem -out device_ec_cert.pem -config x509_config.cfg -subj "/CN=$DEVICE_ID"
+
+          Get-Content device_ec_cert.pem, device_ec_key.pem | Set-Content device_cert_store.pem
+          openssl x509 -noout -fingerprint -in device_ec_cert.pem | % {$_.replace(":", "")} | % {$_.replace("SHA1 Fingerprint=", "")} | Tee-Object fingerprint.txt
+          $fingerprint = Get-Content -Path .\fingerprint.txt
+          
+          # Pass fingerprint to IoTHub 
+          Add-AzIotHubDevice `
+            -ResourceGroupName $ResourceGroupName `
+            -IotHubName $IOTHUB_NAME `
+            -DeviceId $DEVICE_ID `
+            -AuthMethod "x509_thumbprint" `
+            -PrimaryThumbprint $fingerprint`
+            -SecondaryThumbprint $fingerprint
+            
+          # Download Baltimore Cert
+          curl https://cacerts.digicert.com/BaltimoreCyberTrustRoot.crt.pem > $SOURCES_DIR\sdk\samples\iot\BaltimoreCyberTrustRoot.crt.pem
+          
+          # Link IoTHub to DPS service
+          hubConnectionString=$(az iot hub show-connection-string --name $IOTHUB_NAME --key primary --query connectionString -o tsv)
+          az iot dps linked-hub create --dps-name $DPS_NAME --resource-group $ResourceGroupName --connection-string $hubConnectionString --location eastus
+          
+        condition: contains(variables['test_type'], 'iot')
+        env:
+          VCPKG_DEFAULT_TRIPLET: x64-windows-static
+          VCPKG_ROOT: $(Agent.TempDirectory)
+          SOURCES_DIR: $(Build.SourcesDirectory)
+          DEVICE_ID: "aziotbld-c-sample"
```

Comment:
Any reason you are setting these as environment variables as opposed to DevOps variables?

Reply:
Because I don't have access to make them DevOps variables.

Reply:
Unless you mean using the "parameters:" label?

Reply:
You can define variables directly here in this yml file. The values in the matrix entry like `test_type: 'iot'` are variables.